### ISSUE TITLE
remark views content of code syntax as regular Markdown

### ISSUE THREAD

**Issue by TPGPL** [remarkjs/remark-lint#252]
### Subject of the issue

As stated in the title, remark's lint can view content of the code syntax as regular Markdown, which can lead to returning wrong warnings. Seeing the output below, it looks like the problem lies with remark escaping code syntax during linting.

### Your environment

*   **OS**: Windows 10 Pro 20H2
*   **Packages**: [dependencies.txt](https://github.com/remarkjs/remark-lint/files/5541627/dependencies.txt)
*   **Env**: `npm 6.14.8`

### Steps to reproduce

1. **Prepare the work environment.** Create a catalog with an [npm package listing](https://puu.sh/GNE1Y/e336195525.json) and [remark config](http://puu.sh/GNDYF/603b7eb7f4.js)
2. **Install packages with npm.**
3. **Add a test Markdown file.** Here is a [test file with the bug](http://puu.sh/GNE9Z/7b6075bfa7.md)
4. **Use remark to lint the file.**

### Expected behavior

Remark shouldn't return any warnings.

### Actual behavior

```bash
 % remark test.md
# Test Markdown file

| test | test | |
| :- | :- | - |
| test | test | |
| \` | \` | test |
| test row | test row | |

*Test formula: \`a * b * c = abc\`*
test.md
        6:4  warning  Cell should be padded                   table-cell-padding  remark-lint
        6:5  warning  Cell should be padded                   table-cell-padding  remark-lint
   9:1-9:20  warning  Don’t pad `emphasis` with inner spaces  no-inline-padding   remark-lint
  9:23-9:34  warning  Don’t pad `emphasis` with inner spaces  no-inline-padding   remark-lint```

**Comment by wooorm** [comment]
Rendering your [test file](http://puu.sh/GNE9Z/7b6075bfa7.md) on GitHub shows that the `|` in the “code” is not a pipe in code, but actually exits the cell:

# Test Markdown file

| test | test |
| :-- | :-- |
| test | test |
| `|` | test |
| test row | test row |

*Test formula: `a * b * c = abc`*

---

To put a pipe in code in a GFM table, escape it:

```
# Test Markdown file

| test | test |
| :-- | :-- |
| test | test |
| `\|` | test |
| test row | test row |

*Test formula: `a * b * c = abc`*
```

---

You are not running the latest remark, if you experience issues, make sure to update to the latest versions (and add `remark-gfm` when you’re on `remark@13`, as tables changed from being the default to being in a plugin)

---

For math: we don’t support your style of Math (I’m not sure what style that is? 🤔). We do support [math like this if you use `remark-math`](https://github.com/remarkjs/remark-math)
